content/2_Bronze/Intro_DS.mdx

```diff
@@ -710,6 +709,118 @@ See [here](/general/notes-cpp-ds).
+    </Quiz.Answer>
+    <Quiz.Answer>
+      `v.length()`
+      <Quiz.Explanation>
+        Incorrect. The property `length()` does not exist on an `std::vector`
+      </Quiz.Explanation>
+    </Quiz.Answer>
+    <Quiz.Answer>
+      `v.count()` 
+      <Quiz.Explanation>
+        Incorrect. The property `count()` does not exist on an `std::vector`
+      </Quiz.Explanation>
+    </Quiz.Answer>
+    <Quiz.Answer>
+      `v.resize()`
+      <Quiz.Explanation>
+        Incorrect. The property `resize()` resizes the vector to the specified number of elements
+      </Quiz.Explanation>
+    </Quiz.Answer>
+  </Quiz.Question>
+  <Quiz.Question>
+    Which of the following would print out `5`?
+    <Quiz.Answer correct>
+      ```cpp
+      int a = 3, b = 4, c = 5;
+      tuple<int, int, int> t = tie(a, b, c);
+      cout << get<2>(t);
+	  ```
+      <Quiz.Explanation>
+        Correct. `get<i>(t)`: Returns the $i$-th element of the tuple `t`
+      </Quiz.Explanation>
+    </Quiz.Answer>
+    <Quiz.Answer>
+      ```cpp
+      int a = 3, b = 4, c = 5;
+      tuple<int, int, int> t = tie(a, b, c);
+      cout << t.get(2);
+	  ```
+      <Quiz.Explanation>
+        Incorrect. `t.get()` is not a valid method.
+      </Quiz.Explanation>
+    </Quiz.Answer>
+    <Quiz.Answer correct>
+      ```cpp
+      vector<int> v;
+      v.push_back(2);
+      v.push_back(3);
+      v.push_back(7);
+      v.push_back(5);
+      cout << v[3];
+      ```
+      <Quiz.Explanation>
+        Correct. The array is $0$-indexed, which means that `v[3]` corresponds to the "fourth" element in the vector, which is $5$.
+      </Quiz.Explanation>
+    </Quiz.Answer>
+    <Quiz.Answer>
+      ```cpp
+      vector<int> v;
+      v.push_back(2);
+      v.push_back(3);
+      v.push_back(7);
+      v.push_back(5);
+      cout << v[4];
+      ```
+      <Quiz.Explanation>
+        Incorrect. The array is $0$-indexed, which means that `v[3]` corresponds to the "fourth" element in the vector, which is $5$. `v[4]` is out of bounds.
+      </Quiz.Explanation>
+    </Quiz.Answer>
+  </Quiz.Question>
+  <Quiz.Question>
+    Why are data structures useful?
+    <Quiz.Answer correct>
+      Data structures can provide efficient operations on certain types of data.
+      <Quiz.Explanation>
+        
+      </Quiz.Explanation>
+    </Quiz.Answer>
+    <Quiz.Answer correct>
+      Correct. Data structures can store different types of data.
+      <Quiz.Explanation>
+        
+      </Quiz.Explanation>
+    </Quiz.Answer>
+    <Quiz.Answer>
+      They provide a step-by-step set of rules to solve a problem.
+      <Quiz.Explanation>
+        Incorrect. This is an algorithm.
+      </Quiz.Explanation>
+    </Quiz.Answer>
+    <Quiz.Answer>
+      Data structures are sometimes inefficient.
+      <Quiz.Explanation>
+        Incorrect. Data structures are not useful for this reason.
+      </Quiz.Explanation>
+    </Quiz.Answer>
+  </Quiz.Question>
+</Quiz>
+</CPPSection>
+</LanguageSection>
```

Comment:
```suggestion
</CPPSection>
<JavaSection>
<Quiz>
  <Quiz.Question>
    How do you count the number of items in an `ArrayList`? Suppose we named it `list`.
    <Quiz.Answer correct>
      `list.size()`
      <Quiz.Explanation>
        Correct. `size()` returns the number of elements in the data structure!
      </Quiz.Explanation>
    </Quiz.Answer>
    <Quiz.Answer>
      `list.length()`
      <Quiz.Explanation>
        Incorrect. The method `length()` does not exist on an `ArrayList`
      </Quiz.Explanation>
    </Quiz.Answer>
    <Quiz.Answer>
      `list.count()` 
      <Quiz.Explanation>
        Incorrect. The method `count()` does not exist on an `ArrayList`
      </Quiz.Explanation>
    </Quiz.Answer>
    <Quiz.Answer>
      `list.length`
      <Quiz.Explanation>
        Incorrect. The property `length` exists in arrays, but not `ArrayList`s
      </Quiz.Explanation>
    </Quiz.Answer>
  </Quiz.Question>
  <Quiz.Question>
    Why are data structures useful?
    <Quiz.Answer correct>
      Data structures can provide efficient operations on certain types of data.
      <Quiz.Explanation>
        
      </Quiz.Explanation>
    </Quiz.Answer>
    <Quiz.Answer correct>
      Correct. Data structures can store different types of data.
      <Quiz.Explanation>
        
      </Quiz.Explanation>
    </Quiz.Answer>
    <Quiz.Answer>
      They provide a step-by-step set of rules to solve a problem.
      <Quiz.Explanation>
        Incorrect. This is an algorithm.
      </Quiz.Explanation>
    </Quiz.Answer>
    <Quiz.Answer>
      Data structures are sometimes inefficient.
      <Quiz.Explanation>
        Incorrect. Data structures are not useful for this reason.
      </Quiz.Explanation>
    </Quiz.Answer>
  </Quiz.Question>
</Quiz>
</JavaSection>
</LanguageSection>
```

Reply:
^ here's my java code (note i removed q2 because pairs and tuples don't exist). let me know if you have any other ideas for a java question